### PR TITLE
Feat/calculator

### PR DESCRIPTION
**Pull request by mkdoh3**
@itsbaska

### CODE REVIEW

File: lib/calculator.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+class Calculator
+
+    def self.add_values(current_frame, next_frame, third_frame, turn)
+        total = current_frame[0]
+        if(total === 10)
+            return total += current_frame[1] + current_frame[2] if turn === 9
+            total += next_frame[0].to_i
+            return total += third_frame[0].to_i if total === 20 && turn < 8
+            return total += next_frame[1].to_i
+        end
```
Comment (itsbaska) [inline lib/calculator.rb:10]: maybe this can be extracted to #score_first_roll

File: lib/calculator.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        total += current_frame[1]
+        if(total === 10)
+            return total += current_frame[2] if turn === 9
+            return total += next_frame[0]
+        end
```
Comment (itsbaska) [inline lib/calculator.rb]: same here

File: lib/calculator.rb (hunk 1 above)
Comment (itsbaska) [inline lib/calculator.rb]: This i think is now at a good point of extracting.

File: lib/calculator.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+class Calculator
+
+    def self.add_values(current_frame, next_frame, third_frame, turn)
+        total = current_frame[0]
```
Comment (itsbaska) [inline lib/calculator.rb]: lets also be more specific with what total this is. Is this a total for the frame? For the game?

File: lib/calculator.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    def self.spare_value(current_frame, next_frame, turn)
+        return current_frame[2] if turn === 9
+        next_frame[0]
+    end
+
+
+    def self.add_values(current_frame, next_frame, third_frame, turn)
+        frame_total = current_frame[0]
+        return frame_total += strike_values(current_frame, next_frame, third_frame, turn) if frame_total === 10
+        frame_total += current_frame[1]
```
Comment (itsbaska) [inline lib/calculator.rb]: lets now extract these into first roll and second roll

File: lib/calculator.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        return frame_total += spare_value(current_frame, next_frame, turn) if frame_total === 10
+        frame_total
+    end
+
+    def self.score(frames)
+        score = 0
+        frames.each_with_index do |frame, index|
```
Comment (itsbaska) [inline lib/calculator.rb:37]: What was your reasoning behind each_with_index vs just an each?

Comment (mkdoh3) [reply lib/calculator.rb:37]: to access the next frames via index +1 and index + 2

**Comment by mkdoh3** [comment]
..not sure if that's what you meant for first_roll second_roll abstraction

File: lib/calculator.rb (hunk 5 above)
Comment (mkdoh3) [reply lib/calculator.rb:37]: index is also being used to indicate the turn for strike and spare calculations

**Pull request merged**